### PR TITLE
[@lit-labs/router] Convert lit-labs/router to use Wireit

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
Converts the `lit-labs/router` package to [Wireit](https://github.com/google/wireit).

Used https://github.com/lit/lit/pull/2823 as a reference.
Part of https://github.com/lit/lit/issues/2724, to convert the Lit monorepo to Wireit.

Tested by running each command manually, and verifying what outputs are created. As well as Github CI.

I learnt from this PR that `router_test.html` is moved from `src` to `development` via rollup, so rollup is needed as a dep for both dev and production tests. This is now explicitly annotated in the wireit deps.

### CODE REVIEW

**Comment by justinfagnani** [comment]
You should remove the references from tsconfig so that Wireit is in charge of build ordering.

**Comment by AndrewJakubowicz** [comment]
Note: test failure on CI. Investigating. I'm speculating that I've messed up a wireit dependency.

**Comment by AndrewJakubowicz** [comment]
Merged the recent fixes into this PR. It's now ready for review and passing locally for me.

File: packages/labs/router/package.json [hunk 1]
```diff
@@ -41,18 +41,114 @@
...
-    "checksize": "rollup -c --environment=CHECKSIZE",
+    "build": "wireit",
+    "build:rollup": "wireit",
+    "build:ts": "wireit",
+    "build:ts:types": "wireit",
+    "test": "wireit",
+    "test:dev": "wireit",
+    "test:prod": "wireit",
+    "checksize": "wireit",
     "regen-package-lock": "rm -rf node_modules package-lock.json; npm install"
```
Comment (aomarks) [inline packages/labs/router/package.json:52]: Delete `regen-package-lock`

**Review by aomarks — APPROVED** [review]
Awesome!

File: packages/labs/router/package.json (hunk 1 above)
Comment (AndrewJakubowicz) [reply packages/labs/router/package.json]: Done!

**Pull request merged**